### PR TITLE
[IRM] - changes to get incident working in IRM plugin

### PR DESCRIPTION
**Pull request by joeyorlando**
Required for https://github.com/grafana/irm/pull/69

### CODE REVIEW

File: grafana-plugin/pkg/plugin/install.go [hunk 1]
```diff
@@ -104,33 +83,65 @@ func (a *App) handleInstall(w http.ResponseWriter, req *http.Request) {
...
-		}
-		w.WriteHeader(http.StatusBadRequest)
+		// TODO: what to do here?
+		// w.Header().Add("Content-Type", "application/json")
+		// if err := json.NewEncoder(w).Encode(installError); err != nil {
+		// 	log.DefaultLogger.Error("Error encoding response", "error", err)
+		// 	http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		// 	return err, http.StatusInternalServerError
+		// }
+		// w.WriteHeader(http.StatusBadRequest)
```
Comment (joeyorlando) [inline grafana-plugin/pkg/plugin/install.go:93]: TODO

File: grafana-plugin/pkg/plugin/resources.go [hunk 2]
```diff
@@ -121,10 +121,10 @@ func (a *App) handleLegacyInstall(w *responseWriter, req *http.Request) {
 // registerRoutes takes a *http.ServeMux and registers some HTTP handlers.
 func (a *App) registerRoutes(mux *http.ServeMux) {
 	mux.HandleFunc("/plugin/install", a.handleInstall)
```
Comment (joeyorlando) [inline grafana-plugin/pkg/plugin/resources.go:123]: IRM won't need to route to `/plugin/install` (it has it's own install handler)

File: grafana-plugin/pkg/plugin/settings.go [hunk 3]
```diff
@@ -317,32 +317,29 @@ func (a *App) SaveOnCallSettings(settings *OnCallPluginSettings) error {
...
 	startGetSyncData := time.Now()
 	defer func() {
 		elapsed := time.Since(startGetSyncData)
 		log.DefaultLogger.Info("GetSyncData", "time", elapsed.Milliseconds())
 	}()
 
-	onCallPluginSettings, err := a.OnCallSettingsFromContext(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("error getting settings from context = %v", err)
-	}
```
Comment (joeyorlando) [inline grafana-plugin/pkg/plugin/settings.go:330]: I don't think we need to do:
```go
onCallPluginSettings, err := a.OnCallSettingsFromContext(ctx)
```
when `settings *OnCallPluginSettings` is already being passed into `GetSyncData` 🤔

File: grafana-plugin/pkg/plugin/install.go [hunk 4]
```diff
@@ -15,76 +15,55 @@ type OnCallInstall struct {
...
 	if err != nil {
 		log.DefaultLogger.Error("Error request to oncall", "error", err)
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+		return nil, http.StatusInternalServerError, nil, err
 	}
 	defer res.Body.Close()
 
 	if res.StatusCode != http.StatusOK {
-		errorBody, err := io.ReadAll(res.Body)
```
Comment (joeyorlando) [inline grafana-plugin/pkg/plugin/install.go:87]: `err` was never consumed within this conditional block here (IDE warning)

File: grafana-plugin/pkg/plugin/install.go [hunk 5]
```diff
@@ -104,33 +83,66 @@ func (a *App) handleInstall(w http.ResponseWriter, req *http.Request) {
 			}
 		}
 
-		w.Header().Add("Content-Type", "application/json")
-		if err := json.NewEncoder(w).Encode(installError); err != nil {
-			http.Error(w, "Failed to encode response", http.StatusInternalServerError)
-			return
-		}
-		w.WriteHeader(http.StatusBadRequest)
```
Comment (joeyorlando) [inline grafana-plugin/pkg/plugin/install.go:112]: this bit was moved into `handleInstall` handler below (to allow this method to be a bit more generic and be consumed ([here](https://github.com/grafana/irm/pull/69/files#diff-113878fd6f8c6e75911f61bb2c31d3833f905e7cbda9f00d5b795b443e2f200eR32-R46)) by the IRM backend plugin

File: grafana-plugin/pkg/plugin/install.go [hunk 6]
```diff
@@ -104,33 +83,66 @@ func (a *App) handleInstall(w http.ResponseWriter, req *http.Request) {
...
-		if err != nil {
-			log.DefaultLogger.Error("Error saving settings", "error", err)
-			return
+		if persistPluginSettingsToGrafana {
+			pluginSettings.OnCallToken = provisioningData.OnCallToken
+
+			if err = a.SaveOnCallSettings(pluginSettings); err != nil {
+				log.DefaultLogger.Error("Error saving settings", "error", err)
+				return nil, http.StatusInternalServerError, nil, err
+			}
```
Comment (joeyorlando) [inline grafana-plugin/pkg/plugin/install.go:106]: similar comment to [this](https://github.com/grafana/oncall/pull/4987/files#r1744601876)

**Review by matiasb — APPROVED** [review]
Makes sense to me, maybe an extra look from @mderynck would be good to confirm we are not missing anything.

File: grafana-plugin/pkg/plugin/settings.go (hunk 3 above)
Comment (mderynck) [reply grafana-plugin/pkg/plugin/settings.go:330]: 👍 good cleanup to make. This call was leftover from when there was some back and forth about what was coming from where.

**Review by mderynck — APPROVED** [review]

**Pull request merged**